x/gamm/types/pool.go

```diff
@@ -144,27 +144,37 @@ func (params PoolParams) Validate(poolWeights []PoolAsset) error {
 	return nil
 }
 
-func (pa PoolAccount) GetId() uint64 {
+// GetAddress returns the address of a pool.
+// If the pool address is not bech32 valid, it returns an empty address.
```

Comment:
Should we panic instead? Why would a pool not have a poolID (which then implies an address)